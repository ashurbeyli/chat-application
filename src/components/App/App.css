.footer {
    background: #3797d4;
    min-height: 50px;
    position: fixed;
    bottom: 0;
    left: 0;
    width: 100%;
}