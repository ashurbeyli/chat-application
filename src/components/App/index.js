import React from 'react';

import './App.css';

import MessageList from '../MessageList/index';
import MessageForm from '../MessageForm/index';

class App extends React.Component {
  render () {
    return (
      <div>
        <div className="block-container">
          <MessageList/>
        </div>
        <div className="footer">
          <MessageForm/>
        </div>
      </div>
    );
  }
}
export default App;