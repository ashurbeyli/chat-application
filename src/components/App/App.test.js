import React from 'react';
import { shallow } from 'enzyme';

import App from './';

describe('App', () => {
  const app = shallow(<App />);

  it('renders correctly', () => {
    expect(app).toMatchSnapshot();
  });

  it('should render MessageList', () => {
    expect(app.find('.block-container').find('Connect(MessageList)').exists()).toBe(true);
  });

  it('should render MessageForm', () => {
    expect(app.find('.footer').find('Connect(MessageForm)').exists()).toBe(true);
  });

});