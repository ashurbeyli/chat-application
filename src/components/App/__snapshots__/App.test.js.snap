// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="block-container"
>
          <Connect(MessageList) />
</div>,
        <div
          className="footer"
>
          <Connect(MessageForm) />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(MessageList) />,
          "className": "block-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(MessageForm) />,
          "className": "footer",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="block-container"
>
            <Connect(MessageList) />
</div>,
          <div
            className="footer"
>
            <Connect(MessageForm) />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(MessageList) />,
            "className": "block-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(MessageForm) />,
            "className": "footer",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
